linera-execution/src/transaction_tracker.rs

```diff
@@ -149,6 +98,16 @@ impl TransactionTracker {
         Ok(())
     }
 
+    pub fn add_outgoing_messages(
+        &mut self,
+        messages: impl IntoIterator<Item = OutgoingMessage>,
+    ) -> Result<(), ArithmeticError> {
+        for message in messages {
+            self.add_outgoing_message(message)?;
+        }
+        Ok(())
```

Comment:
Could also do `checked_add(messages.len())` here and `outgoing_messages.extend(messages)`?

Reply:
Then we'd need to make `messages` a `Vec` or `impl ExactSizeIterator` which would make it more cumbersome to use. And the performance improvement should be negligible.